src/puppetlabs/puppetdb/http/reports.clj

```diff
@@ -4,30 +4,61 @@
             [net.cgrand.moustache :refer [app]]
             [puppetlabs.puppetdb.http.query :as http-q]
             [puppetlabs.puppetdb.http.events :as e]
-            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json
+                                                    validate-query-params
+                                                    validate-no-query-params
                                                     wrap-with-paging-options]]))
 
+(defn produce-streaming-response
+  "A helper function for produce-streaming-body
+  which adds some defaults for paging-options and
+  does the globals destructuring for us."
+  ([entity version query globals]
+     (produce-streaming-response entity version query {} globals))
+  ([entity version query paging-options globals]
+   (let [{db :scf-read-db url-prefix :url-prefix} globals]
+     (produce-streaming-body entity version query paging-options db url-prefix))))
```

Comment:
if we think it's really important, it seems like the clean approach is to integrate this functionality into produce-streaming body and apply the changes to all http endpoints (which IMO would be a different ticket)